iap/src/main/java/com/limurse/iap/BillingService.kt

```diff
@@ -293,20 +293,21 @@ class BillingService(
                     entry.value?.let {
                         when(it.productType){
                             BillingClient.ProductType.SUBS->{
-                                val firstPricingPhaseList = it.subscriptionOfferDetails?.get(0)?.pricingPhases?.pricingPhaseList?.get(0)
-                                entry.key to DataWrappers.ProductDetails(
-                                    title = it.title,
-                                    description = it.description,
-                                    priceCurrencyCode = firstPricingPhaseList?.priceCurrencyCode,
-                                    price = firstPricingPhaseList?.formattedPrice,
-                                    priceAmount = firstPricingPhaseList?.priceAmountMicros?.div(1000000.0),
-                                    billingCycleCount = firstPricingPhaseList?.billingCycleCount,
-                                    billingPeriod = firstPricingPhaseList?.billingPeriod,
-                                    recurrenceMode = firstPricingPhaseList?.recurrenceMode
-                                )
+                                entry.key to it.subscriptionOfferDetails!![0].pricingPhases.pricingPhaseList.map { pricingPhase ->
```

Comment:
Just for sanity, can we not force unwrap here?

Reply:
Thank you for your feedback. I removed the force unwrap. Also, I moved the obfuscatedAccountId and obfuscatedProfileId from the constructor of `IapConnector` to arguments of `purchase` and `subscribe` to add flexibility.